rtloader/common/builtins/datadog_agent.c

```diff
@@ -653,3 +662,46 @@ static PyObject *set_external_tags(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 
 }
+
+/*! \fn PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+    \brief This function implements the `datadog_agent.obfuscate_sql` method, obfuscating
+    the provided sql string.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a tuple containing the key to retrieve.
+    \return A PyObject* pointer to the value.
+
+    This function is callable as the `datadog_agent.obfuscate_sql` Python method and
+    uses the `cb_obfuscate_sql()` callback to retrieve the value from the agent
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+{
+    // callback must be set
+    if (cb_obfuscate_sql == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    char *rawQuery;
+    if (!PyArg_ParseTuple(args, "s", &rawQuery)) {
+        return NULL;
```

Comment:
You need to release the GIL here since it was acquired line 684

Reply:
Oh whoops, good catch! Updated